aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -200,9 +205,14 @@ std::vector<Tensor> foreach_tensor_norm_cuda(
           const at::cuda::OptionalCUDAGuard device_guard(
               device_of(output_per_tensor));
           auto stream = at::cuda::getCurrentCUDAStream();
+          TensorListMetadata<1> vecResMeta;
+          for (int i = 0; i < ntensors; i++) {
+            vecResMeta.addresses[0][i] =
+                vec_res[i].mutable_data_ptr<scalar_t>();
+          }
           lpnorm_cleanup<scalar_t, NormType::L1><<<ntensors, 512, 0, stream>>>(
               output_per_tensor.const_data_ptr<opmath_t>(),
-              ret_per_tensor.mutable_data_ptr<scalar_t>(),
+              vecResMeta,
```

Comment:
Was `ret_per_tensor.mutable_data_ptr<scalar_t>()` previously a device pointer as `ret_per_tensor` was created with the `options` of `tensors[0]`?
I'm guessing `vecResMeta` is a host pointer here?

Reply:
(was looking at potential sources of IMA during training, which we have observed to go away when this PR was reverted)

Reply:
NVM, looks like passing a struct by value is fine